src/main/java/reactor/netty/resources/PooledConnectionProvider.java

```diff
@@ -741,12 +748,25 @@ public int hashCode() {
 					                   .or((poolable, meta) -> (maxIdleTime != -1 && meta.idleTime() >= maxIdleTime)
 					                           || (maxLifeTime != -1 && meta.lifeTime() >= maxLifeTime)))
 					           .maxPendingAcquire(pendingAcquireMaxCount)
-					           .sizeBetween(0, maxConnections)
 					           .evictInBackground(evictionInterval);
+
+			if (DEFAULT_POOL_GET_PERMITS_SAMPLING_RATE > 0d && DEFAULT_POOL_GET_PERMITS_SAMPLING_RATE < 1d
+					&& DEFAULT_POOL_RETURN_PERMITS_SAMPLING_RATE > 0d && DEFAULT_POOL_RETURN_PERMITS_SAMPLING_RATE < 1d) {
+				poolBuilder = poolBuilder.allocationStrategy(SamplingAllocationStrategy.sizeBetweenWithSampling(
+						0,
+						maxConnections,
+						DEFAULT_POOL_GET_PERMITS_SAMPLING_RATE,
+						DEFAULT_POOL_RETURN_PERMITS_SAMPLING_RATE));
+			}
+			else {
```

Comment:
I would consider at least logging a warning if (some) of the values are set but out of 0-1 bounds (eg. somebody sets sampling rate at 50 thinking it's 50 percent). Negative can probably be silently ignored, and 0 (defaults) totally can be silently ignored, but positives above `1.0` should definitely trigger some warning. That said, maybe that check should be done in a `static final` initializer block upon loading the system property ?

Reply:
fixed